File: phoenix/stock/src/main/java/com/phoenix/dto/ProductDto.java
package com.phoenix.dto;

import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProductDto {
    private String serialNumber;
    private String simNumber;
    private LocalDate checkin;
    private LocalDate checkout;
    private String boxNumber;
    private String brand;
    private String productType;
    private String prodName;
    private String comments;
    private BigDecimal price;
    private boolean returned;
    private boolean returnedstatus;
    private boolean checkedExistence;
    private StockDto stock;
    private AgentProdDto agentProd;
    private AgentProdDto managerProd;
    private AgentProdDto agentwhoSoldProd;
    private AgentProdDto agentReturnedProd;
}

File: unistock/src/app/models/inventory/ProductDto.ts
import { AgentProdDto } from "./AgentProdDto";
import { Stockdto } from "./Stock";

export interface Productdto {
    serialNumber?: string;
    simNumber?: string;
    checkin?: Date;
    checkout?: Date;
    boxNumber?: string;
    brand?: string;
    productType?: string;
    prodName?: string;
    comments?: string;
    price?: number;
    returned?: boolean;
    returnedstatus?: boolean;
    checkedExistence?: boolean;
    stock?: Stockdto;
    agentProd?: AgentProdDto;
    managerProd?: AgentProdDto;
     agentwhoSoldProd?: AgentProdDto;
     agentReturnedProd?: AgentProdDto;

}